Source/Core/Core/PowerPC/Interpreter/Interpreter_Integer.cpp

```diff
@@ -10,10 +10,10 @@
 
 void Interpreter::Helper_UpdateCR0(u32 value)
 {
-  s64 sign_extended = (s64)(s32)value;
-  u64 cr_val = (u64)sign_extended;
-  cr_val = (cr_val & ~(1ull << PowerPC::CR_EMU_SO_BIT)) |
-           ((u64)PowerPC::GetXER_SO() << PowerPC::CR_EMU_SO_BIT);
+  const s64 sign_extended = s64{s32(value)};
```

Comment:
Just wondering, since I see a mix of classic constructors and uniform initialization (going both ways and non-mixed) across this PR, is there any significant difference (or specific reason) why you sometimes use `s64{s32(value)}` (uniform initialization outside, ctor inside), `u32(s32{inst.SIMM_16})` (the opposite; later in this file in `addic`) or just `u32(s32(rGPR[inst.RA]) * inst.SIMM_16)` (only ctor; later in this file in `mulli`)?

At first glance they seem like they should be interchangable in all locations and we could use one syntax for consistency; but I might be missing something...?

Reply:
I use the brace-form where the conversion is not narrowing (expanding SIMM_16 from its bit-field type to `int` doesn't have a loss of information, since it's already a signed type, so it's permissible to use that form of initializing).

On the other hand, if I were to do `u32{SIMM_16}`, this would cause a compiler warning because it's not a lossless conversion (its a sign-conversion), which will cause a compilation warning (and since we build with warnings as errors, it'll cause a compilation error).

It's just a way to allow the compiler to enforce caring about scenarios where conversions aren't lossless (which, in most cases, we actually want here)

Reply:
Got it, thanks 👍